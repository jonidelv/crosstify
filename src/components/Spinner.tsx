import React from 'react'
import ring from '../assets/ring.svg'
import styled from '@emotion/styled'

interface SpinnerProps {
  loading: boolean
  height: number
}

const Spinner: React.FC<SpinnerProps> = ({ loading, height }) => {
  return !loading ? null : <Ring src={ring} height={height || 40} width={height || 40} alt={'Loading Spinner'} />
}

export default React.memo(Spinner)

const Ring = styled.img`
  display: inline-block;
  vertical-align: middle;
`
