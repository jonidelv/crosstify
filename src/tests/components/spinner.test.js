import React from 'react'
import { render } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Spinner from '../../components/Spinner'

describe('test <Spinner /> component', () => {

  it('it renders if loading is true', () => {
    const wrapper = render(
      <Spinner
        loading={true}
      />
    )
    const Ring = wrapper.find('img')

    expect(Ring.length).toBe(1)
    expect(Ring['0'].attribs.width).toBe('40')
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('it set the height and width to 50', () => {
    const wrapper = render(
      <Spinner
        loading={true}
        height={50}
      />
    )
    const Ring = wrapper.find('img')

    expect(Ring.length).toBe(1)
    expect(Ring['0'].attribs.height).toBe('50')
    expect(Ring['0'].attribs.width).toBe('50')
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('not renders if loading is true', () => {
    const wrapper = render(
      <Spinner
        loading={false}
      />
    )
    const Ring = wrapper.find('img')

    expect(Ring.length).toBe(0)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

})
