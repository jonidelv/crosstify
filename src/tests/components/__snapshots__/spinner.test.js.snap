// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Spinner /> component it renders if loading is true 1`] = `
<img
  alt="Loading Spinner"
  class="sc-bdVaJa hbGCZo"
  height="40"
  src="ring.svg"
  width="40"
/>
`;

exports[`test <Spinner /> component it set the height and width to 50 1`] = `
<img
  alt="Loading Spinner"
  class="sc-bdVaJa hbGCZo"
  height="50"
  src="ring.svg"
  width="50"
/>
`;

exports[`test <Spinner /> component not renders if loading is true 1`] = `null`;
